test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -3810,6 +3810,45 @@ public void PackCommand_BuildOutput_DoesNotContainForSpecificFramework(string fr
             }
         }
 
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_BuildOutput_DoesNotContainDefaultExtensions()
+        {
+            // Arrange
+            using (var testDirectory = msbuildFixture.CreateTestDirectory())
+            {
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+                msbuildFixture.CreateDotnetNewProject(testDirectory.Path, projectName);
+
+                using (var stream = new FileStream(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    ProjectFileUtils.SetTargetFrameworkForProject(xml, "TargetFramework", "net5.0");
+                    ProjectFileUtils.AddProperty(xml, "DefaultAllowedOutputExtensionsInPackageBuildOutputFolder", ".dll");
+                    ProjectFileUtils.AddProperty(xml, "GenerateDocumentationFile", "true");
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
+
+                // Act
+                msbuildFixture.PackProject(workingDirectory, projectName, $"-o {workingDirectory}");
+
+                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.0.0.nupkg");
+                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.0.0.nuspec");
+                Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
+                Assert.True(File.Exists(nuspecPath), "The intermediate nuspec file is not in the expected place");
+
+                // Assert
+                using (var nupkgReader = new PackageArchiveReader(nupkgPath))
+                {
+                    var allFiles = nupkgReader.GetFiles().ToList();
+                    Assert.DoesNotContain($"lib/net5.0/{projectName}.xml", allFiles);
```

Comment:
I feel we're only testing one side which is not include `xml` file. Could be able to assert if files with only with `.dll` extension are in package which is more resemble what're trying to achieve?

Reply:
i actually had that earlier but found out that some of the files are always include in the package. as rel.rels, nuspec

Reply:
Another way is add assert that package doesn't include `.dll; .exe; .winmd; .json; .pri` files.
How about that? Then I'll re-trigger the ci for you?

Reply:
> Another way is add assert that package doesn't include .dll; .exe; .winmd; .json; .pri files.
How about that? Then I'll re-trigger the ci for you?

u mean every extension except the .dll ?

Reply:
i added the other doesnot contain tests. can you retrigger the ci

Reply:
Can we still have `Assert.Contain($"lib/net5.0/{projectName}.dll", allFiles);`?

Reply:
This one could be less fragile and more covering. `Assert.False(allFiles.Any(f => f.EndsWith(".xml")));`

Reply:
done, i kept the xml to actually test the removal

Reply:
How about doing same for other types too? Most likely that is all.
Assert.False(allFiles.Any(f => f.EndsWith(".winmd")));
Assert.False(allFiles.Any(f => f.EndsWith(".pri")));
Assert.False(allFiles.Any(f => f.EndsWith(".json")));